### PR TITLE
Patch for PhotoAtWaypoint plugin.  

### PR DESCRIPTION
**Pull request by jediofgever**
https://github.com/ros-planning/navigation2/pull/2041 Introduced PhotoAtWaypoint plugin. 

A directory path should be given to plugin for saving the photos. If the given directory does not exist we create one with `std::experimental::filesystem,` but this may throw exceptions if the given path is an invalid one. So a try catch block should surround that part of the code where we try to create directory. 
This PR has that.

### CODE REVIEW

File: nav2_waypoint_follower/plugins/photo_at_waypoint.cpp [hunk 1]
```diff
@@ -54,22 +54,26 @@ void PhotoAtWaypoint::initialize(
...
+          logger_,
+          "Failed to create directory!: %s required by photo at waypoint plugin, "
+          "exiting the plugin with failure!",
+          save_dir_.c_str()
+        );
+        is_enabled_ = false;
+      }
     }
+  } catch (const std::exception & e) {
+    std::cerr << e.what() << '\n';
```
[the remark this answers is not in the document]
Comment (jediofgever) [reply nav2_waypoint_follower/plugins/photo_at_waypoint.cpp:76]: Hmm unfortunately I have no longer access to my laptop after today(for an indefinite time). Could you take care of this for me please ? 
Thank you

**Comment by SteveMacenski** [comment]
@jediofgever done, I need to open a new PR to merge in those changes since it was on your private fork

**Comment by SteveMacenski** [comment]
See https://github.com/ros-planning/navigation2/pull/2067

**Comment by jediofgever** [comment]
@SteveMacenski , thanks a lot , had some issues with my PC and sent it to service.

**Comment by SteveMacenski** [comment]
Please review #2067

**Pull request closed without merging**